src/containers/orgs/OrgsSagas.js

```diff
@@ -1,36 +1,133 @@
-// import { isValidUUID } from '../../utils/ValidationUtils';
-// import {
-//   GET_RELEVANT_ENTITY_SETS,
-//   SWITCH_ORGANIZATION,
-//   getRelevantEntitySets,
-// } from './OrgsActions';
-//
-// const LOG = new Logger('OrgsSagas');
-//
+/*
+ * @flow
+ */
+
+import {
+  all,
+  call,
+  delay,
+  put,
+  takeEvery,
+  takeLatest,
+} from '@redux-saga/core/effects';
+import {
+  List,
+  Map,
+  fromJS,
+} from 'immutable';
+import {
+  OrganizationsApiActions,
+  OrganizationsApiSagas,
+  PrincipalsApiActions,
+  PrincipalsApiSagas,
+} from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import Logger from '../../utils/Logger';
+import { isValidUUID } from '../../utils/ValidationUtils';
+import {
+  GET_ORGANIZATION_DETAILS,
+  SEARCH_MEMBERS_TO_ADD_TO_ORG,
+  getOrganizationDetails,
+  searchMembersToAddToOrg,
+} from './OrgsActions';
+
+const LOG = new Logger('OrgsSagas');
+
+const {
+  getOrganization,
+  getOrganizationMembers,
+} = OrganizationsApiActions;
+const {
+  getOrganizationWorker,
+  getOrganizationMembersWorker,
+} = OrganizationsApiSagas;
+const { searchAllUsers } = PrincipalsApiActions;
+const { searchAllUsersWorker } = PrincipalsApiSagas;
+
+/*
+ *
+ * OrgsActions.getOrganizationDetails
+ *
+ */
+
+function* getOrganizationDetailsWorker(action :SequenceAction) :Generator<*, *, *> {
+
+  try {
+    yield put(getOrganizationDetails.request(action.id, action.value));
+
+    const organizationId :UUID = action.value;
+    if (!isValidUUID(organizationId)) {
+      throw new Error('organizationId must be a valid UUID');
+    }
+
+    const [orgResponse, orgMembersResponse] = yield all([
+      call(getOrganizationWorker, getOrganization(organizationId)),
+      call(getOrganizationMembersWorker, getOrganizationMembers(organizationId)),
+    ]);
+    if (orgResponse.error) throw orgResponse.error;
+    if (orgMembersResponse.error) throw orgMembersResponse.error;
+
+    const org :Map = fromJS(orgResponse.data);
+    const orgMembers :List = fromJS(orgMembersResponse.data);
+    const orgWithMembers :Map = org.set('members', orgMembers);
+
+    yield put(getOrganizationDetails.success(action.id, orgWithMembers));
+  }
+  catch (error) {
+    LOG.error('getOrganizationDetailsWorker()', error);
+    yield put(getOrganizationDetails.failure(action.id));
+  }
+  finally {
+    yield put(getOrganizationDetails.finally(action.id));
+  }
+}
+
+function* getOrganizationDetailsWatcher() :Generator<*, *, *> {
+
+  yield takeEvery(GET_ORGANIZATION_DETAILS, getOrganizationDetailsWorker);
+}
+
+/*
+ *
+ * OrgsActions.searchMembersToAddToOrg
+ *
+ */
+
+function* searchMembersToAddToOrgWorker(action :SequenceAction) :Generator<*, *, *> {
+
+  try {
+    yield put(searchMembersToAddToOrg.request(action.id, action.value));
+
+    const { organizationId, query } = action.value;
+    if (!isValidUUID(organizationId)) {
+      throw new Error('organizationId must be a valid UUID');
+    }
+
+    yield delay(1000);
```

Comment:
Did you forget to remove this?

Reply:
No, this is intentional. It's there alongside `takeLatest` to avoid firing off a search request for each character typed during member search. It's like a throttle.

Reply:
I believe redux-sagas has first party support for this behavior
https://redux-saga.js.org/docs/api/#throttlems-pattern-saga-args

Reply:
I tried throttle; it didn't work quite the way I wanted. I can demo if you'd like.